ui-lib/src/main/java/co/electriccoin/zcash/ui/MainActivity.kt

```diff
@@ -133,6 +134,35 @@ class MainActivity : ComponentActivity() {
         }
     }
 
+    /**
+     * The body of this function is meat to be execute only while the app run on Firebase Test Lab or
+     * Google Play pre-launch report tests.
+     *
+     * It skips creating a new or restoring an existing wallet by persisting a wallet with a mock seed.
+     * As a result of this, it navigates the monkey test runner from onboarding screens right to the
+     * app home screen.
+     *
+     * @return true if the app is currently being messed with by a monkey test runner and the described
+     * has proceeded or false otherwise.
+     */
+    private fun changeAppFlowForAutomationTests(): Boolean {
+        if (!FirebaseTestLabUtil.isFirebaseTestLab(this))
```

Comment:
It's almost always best to pass `getApplicationContext()` as opposed to the `this` reference.  Doing so prevents ever accidentally leaking the Activity context to clients that usually don't need a reference to the Activity/UI.

From a psuedo-security perspective, it also reduces the receivers ability to cast the context to Activity and do things it shouldn't be doing.

In this case neither is a problem since FirebaseTestLabUtil.isFirebaseTestLab is well behaved, but this is a code smell that always makes me pause and really look closely.

Reply:
Yes, that's true. I've updated it.